### ISSUE TITLE
[BUG] Visual editor Group By on per bucket monitor not showing fields

### ISSUE THREAD

**Issue by llermaly** [opensearch-project/alerting-dashboards-plugin#291]

**Comment by hashworks** [comment]
Same here. Is a template an absolute requirement for "Group by"? It would be great if one could use it for index patterns as well.

**Comment by amsiglan** [comment]
In the GroupBy UI control we allow only fields that have type "keyword". @llermaly  In the above mapping I don't see a field with `keyword` type, hence the empty control.
@qreshi should this be made more apparent in the UX? or should we disable the control when there are no `keyword` type fields?

**Comment by qreshi** [comment]
@amsiglan Yeah, I think we should make the current requirement more apparent in the UX. 

Also the `keyword` filter is because the underlying aggregation is a term aggregation. I believe term aggregations support `keyword, numeric, ip, boolean, or binary` types. So we should test those other types out in the backend to ensure they work as expected and then update the frontend to allow all of those types as well.

**Comment by amsiglan** [comment]
Looked a little more into the mapping shared above and it does have the subfields to specify the `keyword` type for certain fields which I guess should be considered when extracting the fields as part of [`getTypeFromMappings`](https://github.com/opensearch-project/alerting-dashboards-plugin/blob/main/public/pages/CreateMonitor/containers/DefineMonitor/utils/mappings.js#L24).

So in addition to exposing more types, will also check if extracting the subfield type works.

**Comment by BalzGuenat** [comment]
I think I have the same problem. I'm using metricbeat and I'm trying to group a metric by `kubernetes.node.name` (or `kubernetes.node.name.keyword`) and it won't let me. According to `GET logstash-metrics-*/_mapping/`, it should work:

```json
{
  "logstash-metrics-2022.12.21" : {
    "mappings" : {
      "dynamic_templates" : [
        {
          "message_field" : {
            "path_match" : "message",
            "match_mapping_type" : "string",
            "mapping" : {
              "norms" : false,
              "type" : "text"
            }
          }
        },
        {
          "string_fields" : {
            "match" : "*",
            "match_mapping_type" : "string",
            "mapping" : {
              "fields" : {
                "keyword" : {
                  "ignore_above" : 256,
                  "type" : "keyword"
                }
              },
              "norms" : false,
              "type" : "text"
            }
          }
        }
      ],
      "properties" : {
        "@timestamp" : {
          "type" : "date"
        },
        "@version" : {
          "type" : "keyword"
        },
        "kubernetes" : {
          "properties" : {
            "node" : {
              "properties" : {
                "name" : {
                  "type" : "text",
                  "fields" : {
                    "keyword" : {
                      "type" : "keyword",
                      "ignore_above" : 256
                    }
                  },
                  "norms" : false
                }
              }
            }
          }
        }
      }
    }
  }
}

```

**Comment by malro57** [comment]
Hello, 
I'm facing the same problem, any update on the resolution of the bug?
Kind regards,

**Comment by BalzGuenat** [comment]
I was able to work around the issue by constructing a POST request manually.

**Comment by sramli** [comment]
I'm facing the same issue even if my mapping has a couple of keyword fields !!!
any fix or workaround  pls 
![image](https://github.com/opensearch-project/alerting-dashboards-plugin/assets/15628109/114a80c8-1bc9-4f00-97b1-1a2ba86fcfc1)

**Comment by PrakharSharma35** [comment]
> I was able to work around the issue by constructing a POST request manually.

Hi @BalzGuenat i have the same issue as yours was unable to set group by through UI. can u please past an example of POST request u did. i am trying to find one online which uses a group by but unable to find one. Thanks!

**Comment by yoicomment** [comment]
Is there any progress here? Would be nice if this would be fixed/implemented

**Comment by zakisaad** [comment]
This is still an issue, quite a glaring standing omission that used to work just fine. 

We're having to define a lot of our queries manually due to this bug, which is quite cumbersome.

**Comment by nemcikjan** [comment]
any update on this?

**Comment by amsiglan** [comment]
The team has been busy with other commitments, we will pick this up in a week or so, but please feel free to submit a PR with the fix. Thank you for the patience.
